python/python/tests/test_async_client.py

```diff
@@ -3203,6 +3204,120 @@ async def test_type(self, redis_client: TRedisClient):
 
         assert (await redis_client.type(key)).lower() == "none"
 
+    @pytest.mark.parametrize("cluster_mode", [False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_sort_and_sort_store_with_get_or_by_args(
+        self, redis_client: TRedisClient
+    ):
+        key = "{SameSlotKey}" + get_random_string(10)
+        store = "{SameSlotKey}" + get_random_string(10)
+        user_key1, user_key2, user_key3, user_key4, user_key5 = (
+            "user:1",
+            "user:2",
+            "user:3",
+            "user:4",
+            "user:5",
+        )
+
+        # Prepare some data
+        assert await redis_client.hset(user_key1, {"name": "Alice", "age": "30"}) == 2
+        assert await redis_client.hset(user_key2, {"name": "Bob", "age": "25"}) == 2
+        assert await redis_client.hset(user_key3, {"name": "Charlie", "age": "35"}) == 2
+        assert await redis_client.hset(user_key4, {"name": "Dave", "age": "20"}) == 2
+        assert await redis_client.hset(user_key5, {"name": "Eve", "age": "40"}) == 2
+        assert await redis_client.lpush("user_ids", ["5", "4", "3", "2", "1"]) == 5
+
+        if isinstance(redis_client, RedisClient):
+            # Test sort with all arguments
+            assert await redis_client.lpush(key, ["3", "1", "2"]) == 3
+            result = await redis_client.sort(
+                key,
+                limit=Limit(0, 2),
+                get_patterns=["user:*->name"],
+                order=OrderBy.ASC,
+                alpha=True,
+            )
+            assert result == ["Alice", "Bob"]
+
+            # Test sort_store with all arguments
+            sort_store_result = await redis_client.sort_store(
+                key,
+                store,
+                limit=Limit(0, 2),
+                get_patterns=["user:*->name"],
+                order=OrderBy.ASC,
+                alpha=True,
+            )
+            assert sort_store_result == 2
+            sorted_list = await redis_client.lrange(store, 0, -1)
+            assert sorted_list == ["Alice", "Bob"]
+
+            # Test sort with `by` argument
+            result = await redis_client.sort(
+                "user_ids",
+                by_pattern="user:*->age",
+                get_patterns=["user:*->name"],
+                alpha=True,
+            )
+            assert result == ["Dave", "Bob", "Alice", "Charlie", "Eve"]
```

Comment:
Sho has asked me to add tests passing empty lists for list args, can you add a test that passes an empty list for the get_patterns arg?

Reply:
The test with the incorrect data type that Yury mentioned should maybe go in the test below, so that it will be included in both cluster and standalone

Reply:
In cluster we don't have this by/get_pattern so it's irrelevant there.

Reply:
And about the wrong data type, we already test it in the test below as you suggested.